### PR TITLE
Add installation documentation for Emacs

### LINKED ISSUE #393: Add installation instructions for Emacs
I would love to update the website to document the installation instructions for Emacs using lsp-scala (https://github.com/rossabaker/lsp-scala/). I've been planning to do this for awhile but I have limited experience with Emacs so I got stuck on my first attempt and never got around to finish it.

cc/ @rossabaker

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #393 

I'm not too happy with the Emacs lsp-mode client. Most critically,
it doesn't seem to handle diagnostics for multiple files making
it difficult to fix all the compile errors in larger project.

Either way, this is a good start having Emacs documented on the website
will make it easier to upgrade everyone to new releases of Metals and
lsp-mode.

Live demo: https://olafurpg.github.io/metals/docs/editors/emacs.html

### CODE REVIEW

File: docs/editors/emacs.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+First, clone the `lsp-scala` repository.
+
+```sh
+git clone https://github.com/rossabaker/lsp-scala.git
+```
+
+Next, update your Emacs configuration to load `lsp-scala` along with its
+dependencies
+
+```el
```
Comment (olafurpg) [inline docs/editors/emacs.md]: I dumped everything I ended up needing in a clean `init.el` to all the functionality working. It can definitely be improved, and ideally simplified once `lsp-scala` is up on MELPA.

I have no experience with emacs so I suspect this hacky config would be illegal is certain Emacs circles.

Any thoughts @rossabaker?

File: sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala [hunk 2]
```diff
@@ -44,7 +44,7 @@ package scala.meta.metals {
             // Disable fatal warnings so that SemanticDBs are generated even for unused warnings.
             // Down the road, metals can even remove unused imports for you if they are reported
             // as warnings but not if they are errors.
-            "-Xfatal-warning"
+            "-Xfatal-warnings"
```
Comment (olafurpg) [inline sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala]: This is unrelated to Emacs, I caught it while testing out Metals in the http4s repo. sbt-metals disables fatal warning only when exporting the build to bloop, fatal warnings remain enabled in the sbt build.

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 3]
```diff
@@ -206,7 +205,8 @@ class MetalsLanguageServer(
         )
       }
       new InitializeResult(capabilities)
-    }.logError("initialize").asJava
+    }.logError("initialize")
+    timed("initialize")(result).asJava
```
Comment (olafurpg) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala]: Emacs is blocked during the `initialize` request so I delayed all the slow steps to run after the `initialized` notification, this greatly helped speed up startup time on Emacs (I haven't hit the same problem in other editors). We will log from now on how long the `initialize` request takes in order to prevent further regressions, it should only take a few milliseconds to wire together a few components.

File: docs/editors/emacs.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+## Manually trigger build import
+
+To manually trigger a build import, run `M-x lsp-scala-build-import`.
+
+![Import build command](assets/emacs-import-build-command.png)
+
+## Run doctor
+
+Run `M-x lsp-scala-doctor-run` to troubleshoot potential configuration problems
```
Comment (olafurpg) [inline docs/editors/emacs.md]: This command does not exist yet in `lsp-scala`, it would be good to add it https://scalameta.org/metals/docs/editors/new-editor.html#run-doctor

**Comment by fommil** [comment]
> I'm not too happy with the Emacs lsp-mode client

Yeah, it's a huge regression is user friendliness vs the ensime swanky protocol layer. This (combined with the fact that all editors already support the JERKY protocol) is part of the reason why I have never been excited by LSP, particularly for my own use...

**Comment by olafurpg** [comment]
I'm gonna merge this to unblock a followup PR adding BSP server discovery. I opened https://github.com/rossabaker/lsp-scala/pull/12 to add missing commands to lsp-scala that are documented here.

The installation docs will need refinement once lsp-scala is up on MELPA or if maybe eglot (https://github.com/joaotavora/eglot) ends up solving the issues of lsp-mode. But for now I believe this is a good step forward.

@fommil I'm not surprised if lsp-mode is a regression compared to ENSIME, the issues I hit on yesterday make the UX quite sub-optimal. I'm optimistic however that LSP clients for Emacs will improve with time :)

**Comment by fommil** [comment]
@olafurpg 

> I'm optimistic however that LSP clients for Emacs will improve with time :)

I believe the same was said 12 months ago :wink:
